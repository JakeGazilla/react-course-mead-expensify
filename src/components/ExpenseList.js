import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {
      props.expenses.length === 0 ? (
        <p>No expenses saved</p>
      ) : (
        <ul>
          {props.expenses.map((expense) => {
            return <ExpenseListItem {...expense} key={expense.id}/>
          })}
        </ul>
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
}

export default connect(mapStateToProps)(ExpenseList);






// longer way to do it

// const ConnectedExpenseList = connect((state) => {
//   return {
//     expenses: state.expenses
//   };
// })(ExpenseList);
// export default ConnectedExpenseList;