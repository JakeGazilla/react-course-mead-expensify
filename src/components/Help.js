import React from 'react';

const HelpPage = () => (
  <div>
    This is from your help ass component
  </div>
);

export default HelpPage;