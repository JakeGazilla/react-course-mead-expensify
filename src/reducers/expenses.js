// expenses reducer

const expensesReducerDefaultState = [];
const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
    // console.log(state);
    console.log(action.type)
      return [
        ...state,
        action.expense
      ];
    case 'REMOVE_EXPENSE':
    console.log(action.type)
      return state.filter(({id}) => id !== action.id )
    case 'EDIT_EXPENSE':
    console.log(action.type)
      return state.map((expense) => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.updates
          };
        } else {
          return expense;
        }
      });
    default: 
      return state;
  }
};

export default expensesReducer;