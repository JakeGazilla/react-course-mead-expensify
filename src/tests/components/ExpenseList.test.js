import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/fixtures';

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses}/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should render ExpenseList with empty msg', () => {
  const wrapper = shallow(<ExpenseList expenses={[]}/>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});