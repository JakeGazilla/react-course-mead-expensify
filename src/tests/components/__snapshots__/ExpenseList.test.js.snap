// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty msg 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <p>
    No expenses saved
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <h1>
    Expense List
  </h1>
  <ul>
    <ExpenseListItem
      amount={111}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={109500}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </ul>
</div>
`;
